libs/blocks/merch/checkout-context.js

```diff
@@ -0,0 +1,30 @@
+const DEFAULT_WORKFLOW_STEP = 'email';
+const DEFAULT_CLIENT_ID = 'adobe_com';
+
+function getWorkflowStep(searchParams, checkoutContext) {
+  const workflowStep = searchParams?.get('workflowStep') ?? checkoutContext?.workflowStep ?? DEFAULT_WORKFLOW_STEP;
+  return workflowStep?.replace('_', '/');
+}
+
+/**
+ * Checkout Type, Workflow Step and Market Segment can be set in the OST link.
+ * If there is no value in search params, value is taken from global config.
+ * If there is no value defined in global config, default value is used.
+ * @param {*} searchParams link level overrides for checkout parameters
+ * @returns checkout context object required to build a checkout url
+ */
+export default function getCheckoutContext(searchParams, config) {
+  const { commerce } = config;
```

Comment:
I'm wondering if config is really the right place for this cta configuration . Won't things like the checkoutContext be decided on a page by page basis? If so, these configurations should go in the metadata, and not the project config

Reply:
>Won't things like the checkoutContext be decided on a page-by-page basis?
this was the case for Dexter indeed. However, in Milo, we are challenging the actual need for this requirement.
These parameters can be global + overwritten on the link level. 

We will need to use metadata for 'promo code', but this parameter is out of the scope of this PR.
Another possible use case for metadata might be the 'checkoutType' parameter, but it also can be that we will switch all CTAs at once, so there is no need to make it configurable in metadata right now.

Reply:
Sure, but by that logic there's also no need to make it configurable over the config object either right now. If you don't want to make it configurable, then I'd least switch your code to check the page metadata instead of the config object.

Reply:
> then I'd least switch your code to check the page metadata instead of the config object.
added this for checkoutType